eng/templates/jobs/build-from-source.yml

```diff
@@ -0,0 +1,138 @@
+parameters:
+  # Specifies the prefix for source-build jobs added to pipeline. Use this if disambiguation is necessary.
+  jobNamePrefix: 'Build_From_Source'
+
+  # Specifies that the repo only builds managed components. This indicates that output is not RID-specific 
+  # and only one build job is required. A linux x64 machine will be used, but the build must continue to 
+  # work on all supported platforms equally.
+  # If false, specifies that the repo has native components that can link to platform-provided libraries.
+  #   - Adds additional portable jobs for supported portable platforms
+  #   - Adds additional non-portable jobs for supported non-portable platforms
+  #     - The job passes /p:ArcadeBuildNonPortable=true to the build to enable non-portable behavior.
+  #     - The job passes a platform-specific TargetRid MSBuild property.
+  managedOnly: false
+  
+  # Specifies whether prebuilt checks are enforced. When true, if prebuilts are introduced, the build job will fail.
+  prebuiltCheck: true
+
+  # Defines the platforms on which to run each build job
+  # Note: This parameter is not expected to be set by the consumer of this template
+  platforms:
+
+  # Always run centos71 portable build
+  - name: 'Centos71_Portable'
+    include: 'always'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-3e800f1-20190501005343'
+  
+  # If managedOnly is false, run a macOS portable build
+  - name: 'MacOS_Portable'
+    jobProperties:
+      pool:
+        name: Hosted macOS
+
+  # Non-portable platforms
+  - name: 'Centos71'
+    targetRID: 'centos.7-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-3e800f1-20190501005343'
+
+  - name: 'Centos8'
+    targetRID: 'centos.8-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-8-daa5116-20200325130212'
+
+  - name: 'Debian9'
+    targetRID: 'debian.9-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:debian-stretch-d61254f-20190807161114'
+
+  - name: 'Fedora30'
+    targetRID: 'fedora.30-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-30-38e0f29-20191126135223'
+
+  - name: 'Fedora32'
+    targetRID: 'fedora.32-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-32-20200512010618-163ed2a'
+
+  - name: 'MacOS'
+    targetRID: 'osx.10.14-x64'
+    jobProperties:
+      pool:
+        name: Hosted macOS
+
+  - name: 'Ubuntu1804'
+    targetRID: 'ubuntu.18.04-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-f90bc20-20180320154721'
+
+jobs:
+- job: Build_From_Source_Main
+  displayName: Build From Source Main
```

Comment:
I see I suggested "main", but not sure why... now that I see it here, I think we should go with `Complete` to make that string more obvious when it shows up in a list of job dependencies.

On the display name side, I'm also wondering if "Source-build (x)" would be more concise and if we should stick with that name in "branding" on the source-build effort. Seems more familiar, e.g. the effort's called arcade-powered source-build.

![image](https://user-images.githubusercontent.com/12819531/91493360-d2e47e00-e87c-11ea-822a-dd4d31532497.png)

![image](https://user-images.githubusercontent.com/12819531/91493389-dc6de600-e87c-11ea-8e02-cc9b91662d27.png)

Reply:
That makes sense.  What do you think about the name of the script itself?  `build-from-source.yml` -> `source-build.yml`?